src/components/Page/components/Header/Header.tsx

```diff
@@ -45,7 +46,7 @@ export interface HeaderProps extends TitleProps {
   /** Collection of breadcrumbs */
   breadcrumbs?: BreadcrumbsProps['breadcrumbs'];
   /** Collection of secondary page-level actions */
-  secondaryActions?: MenuActionDescriptor[];
+  secondaryActions?: MenuActionDescriptor[] | React.ReactNode;
```

Comment:
[question] given that the prop is plural, should this be a list of `React.ReactNode` instead of a single node?

Reply:
Great question!

When passing an array of `ReactNode`s, Polaris `<Page />`, `<Header />` and `<MenuAction />` components would have to handle the appearance of this list of components passed to it. And this is where it get tricky - `ReactNode` is kind of a black hole that’s only being used if none of the default `secondaryActions: MenuActionDescriptor[]` options work for the consumer (_which is rare_). So in that case, I’m thinking that allowing the consumer to take care of whatever custom actions they need to pass and how they want them to appear instead of giving them only _some_ flexibility by passing `ReactNode` but at the same time adding constraints by requiring it to be an array and as a result having to process/style the list on Polaris side and collapse them into actions dropdown when it’s too many and going with the assumption that regardless of what kind of nodes consumers pass to it would work with the way actions are handled on Polaris side.

As for the prop name being plural - I can see how it may seem not intuitive, at the same time it’s up to consumers how many custom react nodes they want to pass to it, so it still could be multiple, could be one, and if we take `children` prop as an example - it’s also plural and of type `React.ReactNode` and does the same thing.

So I’m leaning towards having the secondary actions API like this: 

- [CURRENT] Strict API for `MenuActionDescriptor[]` actions that are being handled on Polaris side, laid out, collapsed on mobile, etc.
- [NEW] Open API to fully shift the appearance and rendering logic responsibility of actions to consumers if they desire something custom and default `MenuActionDescriptor[]` actions don’t work for them